src/NHibernate/Loader/Hql/QueryLoader.cs

```diff
@@ -397,6 +397,14 @@ private void CheckQuery(QueryParameters queryParameters)
 			{
 				throw new QueryException("ResultTransformer is not allowed for 'select new' queries.");
 			}
+
+			if (Factory.Settings.IsQueryCacheEnabled && queryParameters.Cacheable)
+			{
+				foreach (var persister in _entityPersisters.Where(x => !x.HasUpdateTimestampsCache))
+				{
+					throw new QueryException($"Never cached entity:{persister.EntityName} cannot be used in cacheable query");
```

Comment:
By throwing inside a `foreach` statement, only the first entity name will be included in the exception. Here we should concatenate the entity names before throwing the exception.

Reply:
with entityPersistersSpaces and collectionPersistersSpaces ok